src/NTP/Util.hs

```diff
@@ -64,3 +69,14 @@ selectIPv6 = find (\a -> addrFamily a == AF_INET6)
 
 selectIPv4 :: [AddrInfo] -> Maybe AddrInfo
 selectIPv4 = find (\a -> addrFamily a == AF_INET)
+
+withSocketsDoLifted :: (MonadIO m, MonadMask m) => m a -> m a
+withSocketsDoLifted action = do
+    -- `MonadBaseControl IO` is not provided sometimes, so hack with threads
+    -- is used
+    bracket (liftIO $ newTVarIO False)
+            (liftIO . atomically . flip writeTVar True) $
+            \exited -> do
+                liftIO . void . forkIO $
```

Comment:
Well, it affects behaviour, because implementation creates new thread. I think this can be relevant, because you're not just adding type constraints and not only adding `liftIO`.
Haddock is just good doc, why not write more docs? But at least I want such thoughts to be written so somebody can look at this later and understand, what's going on.